### ISSUE TITLE
Build failure with GHC 9.2 alpha2

### ISSUE THREAD

**Issue by andreasabel** [haskell-compat/base-compat#74]
Didn't find an issue for this yet:
```
ghc-9.2.0.20210422$ cabal install base-compat --allow-newer
...
[ 63 of 118] Compiling Data.Semigroup.Compat ( src/Data/Semigroup/Compat.hs, dist/build/Data/Semigroup/Compat.o, dist/build/Data/Semigroup/Compat.dyn_o )

src/Data/Semigroup/Compat.hs:25:5: error:
    Not in scope: type constructor or class ‘Option’
   |
25 |   , Option(..)
   |     ^^^^^^^^^^

src/Data/Semigroup/Compat.hs:26:5: error: Not in scope: ‘option’
   |
26 |   , option
   |     ^^^^^^
cabal: Failed to build base-compat-0.11.2.
```

**Comment by phadej** [comment]
fixed in `master` already https://github.com/haskell-compat/base-compat/commit/d2cd678974dea75ee402a0f7b78e7f87ed995b45, see `base-4.16` branch.

**Comment by andreasabel** [comment]
> perhaps around the time of the release candidate being announced

GHC 9.2.1 RC1 is out now, and I am testing my projects with it.  Many depend on `base-compat`, so having a release on hackage would be great!

**Comment by phadej** [comment]
At the time of writing I don't see any announcements, nor even tag of 9.2.1 (FWIW, GHC releases "happen" only when they are announced on the mailing lists). Please be patient.

**Comment by andreasabel** [comment]
OK.  I found it here: https://downloads.haskell.org/ghc/9.2.1-rc1/

**Comment by RyanGlScott** [comment]
Sorry for the delay on this. Indeed, I was waiting for an official announcement of GHC 9.2.1-rc1, which happened [just recently](https://discourse.haskell.org/t/ghc-9-2-1-rc1-is-now-available/2915). I'll try to make a new `base-compat` release soon.

**Comment by RyanGlScott** [comment]
The `master` branch of `base-compat` is essentially in a releasable state, although I just discovered that there is at least one last-minute change to `Data.Tuple` that would affect its export list (and therefore affect `base-compat`)—see [here](https://gitlab.haskell.org/ghc/ghc/-/merge_requests/6407#note_372664). I don't have a sense for how long it would take for that change to make its way into a GHC bindist, and moreover, we could add that change to `base-compat` later without needing a major version bump. For that reason, I'm somewhat inclined to make a new release of `base-compat` now and adapt to subsequent changes later.

**Comment by RyanGlScott** [comment]
> It would make some sense to make [`OneTuple`](https://hackage.haskell.org/package/OneTuple) a compat package providing `Solo` for older GHCs. Then `base-compat-batteries` could use it. If that sounds good, we probably need a separate issue for coordination of that plan.

I'd be fine with such a plan, if you'd like to take the lead in implementing it.

In any case, it sounds like we'll put off dealing with `Solo` for now. In light of this, I've released `base-compat{-batteries}-0.12.0`, so I'll go ahead and close this issue. Do re-open if there is something I've overlooked.

**Comment by andreasabel** [comment]
@RyanGlScott @phadej : Thanks everyone!
